patoolib/util.py

```diff
@@ -398,6 +398,11 @@ def guess_mime_file_text (file_prog, filename):
 
 def check_existing_filename (filename, onlyfiles=True):
```

Comment:
Since we're changing this method to actually check for either existing file name or existing directory, let's change the variable names accordingly:
`check_existing_filename`  -> `check_existing_path`
`filename` -> `path`
`Ensure that given filename is a valid, existing file.` -> `Ensure that given path is valid.`